target/linux/kirkwood/modules.mk

```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (C) 2006-2014 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define KernelPackage/hwmon-tc654
```

Comment:
Why it's target specific package? It could be used in different devices too.

Reply:
There is a clash with the apm821xx target (ping @chunkeey for opinions). It has SENSORS_TC654 compiled into the kernel (and the tc654 driver has patches added in apm821xx/patches-5.15/010-hwmon-tc654-add-thermal_cooling-device.patch) . 

I agree that the target-specific hwmon-tc654 could/should be moved to be a generic one, but then the apm821xx target would also need to be updated to use the module, or I also have to compile in the tc654 driver - which means that the driver would be enabled for the whole kirkwood target as there are no subtargets there. Given that most if not all kirkwood boards are NAND-based, the size increase would likely be close to be taken as a rounding error, so I'm okay with either.

Reply:
@wigyori @adschm  It would be great to have this PR accepted soon as the Legacy Stora software has not been receiving any security updates for a few years now.

Reply:
clash? Do you have a build failure or something? 

From what I know, the kmod-hwmon-tc654 will just not contain a .ko for the apm821xx. But this there's otherwise no real-life consequences. 

The NETGEAR WNDR4700 SoC is running hotter because it has a 200MHz higher clock than any other APM821xx device. The thermal management is handled by the thermal-zone framework itself, that's why it's "neat" to have the fan-controller+temperature sensor built-in.